ConfigurationSystem/private/ServiceInterface.py

```diff
@@ -16,250 +18,247 @@
+    gLogger.info("New version it is!")
     return retVal
 
-  def getCompressedConfigurationData( self ):
+  def getCompressedConfigurationData(self):
     return gConfigurationData.getCompressedData()
 
-  def getVersion( self ):
+  def getVersion(self):
     return gConfigurationData.getVersion()
 
-  def getCommitHistory( self ):
-    files = self.__getCfgBackups( gConfigurationData.getBackupDir() )
-    backups = [ ".".join( fileName.split( "." )[1:-1] ).split( "@" ) for fileName in files ]
+  def getCommitHistory(self):
+    files = self.__getCfgBackups(gConfigurationData.getBackupDir())
+    backups = [".".join(fileName.split(".")[1:-1]).split("@") for fileName in files]
     return backups
 
-  def run( self ):
+  def run(self):
     while True:
       iWaitTime = gConfigurationData.getSlavesGraceTime()
-      time.sleep( iWaitTime )
+      time.sleep(iWaitTime)
       self.__checkSlavesStatus()
 
-  def getVersionContents( self, date ):
+  def getVersionContents(self, date):
     backupDir = gConfigurationData.getBackupDir()
-    files = self.__getCfgBackups( backupDir, date )
+    files = self.__getCfgBackups(backupDir, date)
     for fileName in files:
-      with zipfile.ZipFile( "%s/%s" % ( backupDir, fileName ), "r" ) as zFile:
+      with zipfile.ZipFile("%s/%s" % (backupDir, fileName), "r") as zFile:
         cfgName = zFile.namelist()[0]
-        retVal = S_OK( zlib.compress( zFile.read( cfgName ) , 9 ) )
+        retVal = S_OK(zlib.compress(zFile.read(cfgName), 9))
       return retVal
-    return S_ERROR( "Version %s does not exist" % date )
+    return S_ERROR("Version %s does not exist" % date)
 
-  def __getCfgBackups( self, basePath, date = "", subPath = "" ):
-    rs = re.compile( r"^%s\..*%s.*\.zip$" % ( gConfigurationData.getName(), date ) )
-    fsEntries = os.listdir( "%s/%s" % ( basePath, subPath ) )
-    fsEntries.sort( reverse = True )
+  def __getCfgBackups(self, basePath, date="", subPath=""):
+    rs = re.compile(r"^%s\..*%s.*\.zip$" % (gConfigurationData.getName(), date))
+    fsEntries = os.listdir("%s/%s" % (basePath, subPath))
+    fsEntries.sort(reverse=True)
     backupsList = []
     for entry in fsEntries:
-      entryPath = "%s/%s/%s" % ( basePath, subPath, entry )
-      if os.path.isdir( entryPath ):
-        backupsList.extend( self.__getCfgBackups( basePath, date, "%s/%s" % ( subPath, entry ) ) )
-      elif os.path.isfile( entryPath ):
-        if rs.search( entry ):
-          backupsList.append( "%s/%s" % ( subPath, entry ) )
+      entryPath = "%s/%s/%s" % (basePath, subPath, entry)
+      if os.path.isdir(entryPath):
+        backupsList.extend(self.__getCfgBackups(basePath, date, "%s/%s" % (subPath, entry)))
+      elif os.path.isfile(entryPath):
+        if rs.search(entry):
+          backupsList.append("%s/%s" % (subPath, entry))
     return backupsList
 
-  def __getPreviousCFG( self, oRemoteConfData ):
-    remoteExpectedVersion = oRemoteConfData.getVersion()
```

Comment:
This changes the semantics.Now the version was evaluated once at the moment the function definition was processed. Previously, the version was reevaluated before each function call. And I think in this case this really matter.

Reply:
I am not sure I understand: why would it matter? I can restore it, no problem, but I would like to understand what's the reason, as `remoteExpectedVersion` is not used.

Reply:
Yes, you are right, I thought the subsequent call was using this variable anticipating the developer intentions. And I think this will be indeed correct to get the value into the variable and pass it to the subsequent self.__getCfgBackups(... call.